app/src/org/commcare/android/javarosa/IntentCallout.java

```diff
@@ -214,8 +214,8 @@ private boolean processOdkResponse(Intent intent, TreeReference intentQuestionRe
 
                 // Get our response value
                 String responseValue = response.getString(key);
-                if (key == null) {
```

Comment:
I don't see any point of resetting null keys to empty string. It looks like the intention here was to do this check on value instead of key and someone just made a mistake while coding.

Reply:
Agreed. In the original code that got pulled over the intent is a smidge unclear, but it's hard to believe that it was intended to function the way it was written.